Makefile

```diff
@@ -46,16 +49,47 @@ develop: publish-build
 	docker-compose logs -f
 
 # Clean
+clean: TAG := latest
 clean:
 	docker-compose stop
 	docker-compose down --remove-orphans --volumes --rmi all
 	docker-compose rm -svf
-	docker rmi --force openwisp/openwisp-base:latest \
-				openwisp/openwisp-base:intermedia-system \
-				openwisp/openwisp-base:intermedia-python \
+	docker rmi --force $(DOCKER_USERNAME)/openwisp-base:$(TAG) \
+				$(DOCKER_USERNAME)/openwisp-base:intermedia-system \
+				$(DOCKER_USERNAME)/openwisp-base:intermedia-python \
+				`docker images -f "dangling=true" -q` || true
+
+release-clean: clean
+	docker-compose rm -svf
+	docker rmi --force $(DOCKER_USERNAME)/openwisp-base:$(TAG) \
+				$(DOCKER_USERNAME)/openwisp-base:intermedia-system \
+				$(DOCKER_USERNAME)/openwisp-base:intermedia-python \
 				`docker images -f "dangling=true" -q` || true
 
 # Publish
+publish: TAG := latest
 publish: publish-build develop-runtests
-	docker push openwisp/openwisp-base:latest
+	docker login -u $(DOCKER_USERNAME)
+	docker push $(DOCKER_USERNAME)/openwisp-base:$(TAG)
+	docker-compose push
+
+release-publish: publish
+	docker push $(DOCKER_USERNAME)/openwisp-base:$(TAG)
+	docker-compose push
+
+travis-publish: TAG := nightly
+travis-publish: travis-runtests
+	echo "$(DOCKER_PASSWORD)" | docker login -u "$(DOCKER_USERNAME)" --password-stdin
+	docker push $(DOCKER_USERNAME)/openwisp-base:$(TAG)
 	docker-compose push
+	echo "$(GITHUB_TOKEN)" | docker login docker.pkg.github.com -u $(GITHUB_USERNAME) --password-stdin
+	docker tag $(DOCKER_USERNAME)/openwisp-base:$(TAG) docker.pkg.github.com/$(GITHUB_USERNAME)/docker-openwisp/openwisp-base:$(TAG)
+	docker push docker.pkg.github.com/$(GITHUB_USERNAME)/docker-openwisp/openwisp-base:$(TAG)
+	for img in $$(docker-compose config | awk '{if($$0 ~ /image: $(DOCKER_USERNAME)/ && !visited[$$0]++) print $$2}'); do \
```

Comment:
Do you mean readibility in the travis logs itself or the thing in the modified line 88 (for img in ...)?
If it's travis, yeah I do agree. Should I make the docker login and docker push output to `/dev/null`?
If it's about that modified line 88, well it's important because you have to tag all of those images again. I will think of another approach

Reply:
No, I just mean that it's harder to read this part, can use you spaces and `\` character to make part look clean, not just line `88` but couple of lines around it as well. :smile:

Reply:
Should `docker.pkg.github.com` be a variable?  Someone may want to use Gitlab or some other private registry.

Reply:
@2stacks I don't think someone will be using `travis-publish`, besides `publish` is only supported for docker hub only :smile: 
The task isn't require me to add that functionality, but if you guys want me to add it, then I can do it but it'll take more time :sweat_smile:

Reply:
>besides publish is only supported for docker hub only

That doesn't make sense to me.  You have a separate push command for docker hub and git hub.  I was asking if this can be reduced to a single command that can support dockerhub, github, gitlab and custom registries.  They are all using the docker api so a single command "should" work for all of them.

Reply:
Sorry, yeah now it's supported, I forgot to say in my current commit that this is possible, but I already did change the README so user will notice it :sweat_smile: